File: 1/1000s/15s/1508/RangeSumOfSortedSubarraySums.cpp
/*
https://leetcode.com/problems/range-sum-of-sorted-subarray-sums/description/?envType=daily-question&envId=2024-08-04s
*/

class Solution {
public:
    int rangeSum(vector<int>& nums, int n, int left, int right) {
        vector<long long>allSums;
        int sz=nums.size(), mod=1e9+7;
        long long res=0;
        for(int i=0; i<sz; ++i){
            long long cursm=0;
            for(int j=i; j<sz; ++j){
                cursm+=nums[j];
                cursm%=mod;
                allSums.push_back(cursm);
            }
        }
        sort(allSums.begin(), allSums.end());
        for(int i=left-1; i<right; ++i){
            res+=allSums[i];
            res%=mod;
        }
        return res;
    }
};

File: 1/1000s/15s/1508/RangeSumOfSortedSubarraySums.py
# https://leetcode.com/problems/range-sum-of-sorted-subarray-sums/description/?envType=daily-question&envId=2024-08-04s

class Solution:
    def rangeSum(self, nums: List[int], n: int, left: int, right: int) -> int:
        allSubArraySums=[]
        lengthOfNums=len(nums)
        modulus=1000000000+7
        resultantSum=0
        for i in range(0, lengthOfNums, 1):
            currentSubArraySum=0
            for j in range(i, lengthOfNums, 1):
                currentSubArraySum+=nums[j]
                currentSubArraySum%=modulus
                allSubArraySums.append(currentSubArraySum)
        allSubArraySums.sort()
        for i in range(left-1, right, 1):
            resultantSum+=allSubArraySums[i]
            resultantSum%=modulus
        return resultantSum